custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/provider.go

```diff
@@ -0,0 +1,59 @@
+package coreprovider
+
+import (
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/metrics/pkg/apis/metrics"
+	"sigs.k8s.io/metrics-server/pkg/api"
+)
+
+// CoreProvider is a provider of core metrics.
+type CoreProvider struct{}
+
+var _ api.MetricsGetter = &CoreProvider{}
```

Comment:
what does this line do? Triggers some inits?

Reply:
It checks (during compilation time) if CoreProvider implements api.MetricsGetter inferface.

Reply:
nice :-)